### PR TITLE
ws.inactivity.timeout populates http.keealive.timeout

### PR DESCRIPTION
**Pull request by jitsni**
https://github.com/kaazing/roadmap/issues/683
- ws.inactivity.timeout populates http.keepalive.timeout (if not set)
- warns when http.keepalive.timeout is less than ws.inactivity.timeout as that may close connections
- Cleaning of the options code
- TODO the options map can be common to both accept and connect. Now just retained whatever was done.

### CODE REVIEW

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java [hunk 1]
```diff
@@ -72,22 +72,20 @@ public void setAllowedServices() {
...
-    @Ignore("XSD no longer validates accept-options types")
     public void testHttpKeepAliveTimeoutOption() throws Exception {
         // expect default if 0 is specified
         expectSuccess("http.keepalive.timeout", "0 minutes", "http[http/1.1].keepAliveTimeout", 30);
 
         expectSuccess("http.keepalive.timeout", "10 seconds", "http[http/1.1].keepAliveTimeout", 10);
         expectSuccess("http.keepalive.timeout", "10 minutes", "http[http/1.1].keepAliveTimeout", 600);
         expectSuccess("http.keepalive.timeout", "0.5 minutes", "http[http/1.1].keepAliveTimeout", 30);
-        expectSuccess("http.keepalive.timeout", "", "http[http/1.1].keepAliveTimeout", 30);
+        //expectSuccess("http.keepalive.timeout", "", "http[http/1.1].keepAliveTimeout", 30);
```
Comment (dpwspoon) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java]: What is going on here? if set to empty should we through an error or set the default?

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java [hunk 2]
```diff
@@ -134,6 +132,15 @@ public void testTcpMaximumOutboundRateOption() throws Exception {
     }
 
     @Test
+    public void testWsInactivityTimeout() throws Exception {
+        expectSuccess("ws.inactivity.timeout", "60", "ws.inactivityTimeout", 60000L, "http[http/1.1].keepAliveTimeout", 60);
+        expectSuccess("ws.inactivity.timeout", "60s", "ws.inactivityTimeout", 60000L, "http[http/1.1].keepAliveTimeout", 60);
+        //https://github.com/kaazing/gateway/issues/595
+        //expectSuccess("ws.inactivity.timeout", "60000ms", "ws.inactivityTimeout", 60000L, "http[http/1.1].keepAliveTimeout", 60);
```
Comment (dpwspoon) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java]: Good find!

**Comment by dpwspoon** [comment]
Is this ready to merge?

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java (hunk 1 above)
Comment (jitsni) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/AcceptOptionsTest.java]: I just enabled the ignored test. I don't know the earlier behaviour and let us deal with it as a separate PR

**Comment by jitsni** [comment]
Yes, ready for merge

**Pull request merged**